WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreName/StoreNameSetupViewModel.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import Yosemite
+
+/// View model for `StoreNameSetupView`.
+///
+final class StoreNameSetupViewModel: ObservableObject {
+
+    @Published var name: String
+    @Published private(set) var isSavingInProgress = false
+    @Published private(set) var errorMessage: String?
+
+    /// Whether the Save button should be enabled
+    var shouldEnableSaving: Bool {
+        name.isNotEmpty && name != initialStoreName
+    }
+
+    private let siteID: Int64
+    private let stores: StoresManager
+    private let onNameSaved: () -> Void
+    private let initialStoreName: String
+
+    init(siteID: Int64,
+         name: String,
+         stores: StoresManager = ServiceLocator.stores,
+         onNameSaved: @escaping () -> Void) {
+        self.siteID = siteID
+        self.name = name
+        self.initialStoreName = name
+        self.stores = stores
+        self.onNameSaved = onNameSaved
+    }
+
+    func saveName() {
+        errorMessage = nil
+        isSavingInProgress = true
+        stores.dispatch(SiteAction.updateSiteTitle(siteID: siteID, title: name, completion: { [weak self] result in
```

Comment:
Opinion: IMO, extracting the dispatch part into a private method and making it async would make this more readable. (Or look neater 😅)

We can await the result and handle it using a switch case here.

Reply:
Split the code in 28e89fc10cb5814a3eb58f5ad8f02de74eddc42a.